### PR TITLE
io: add separate embedded-io-adapters crate, add futures/tokio support.

### PR DESCRIPTION
**Pull request by Dirbaio**
If we want to release `embedded-io` 1.0 and the adapters live there, we can't make any 
breaking changes to them after release. Putting them in a separate crate 
allows doing breaking changes without breaking everyone using the IO traits.

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
Other than the CI thing, looks good to me.
Thank you!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 on:
   push: # Run CI for all branches except GitHub merge queue tmp branches
-    branches-ignore:
-    - "gh-readonly-queue/**"
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (Dirbaio) [reply .github/workflows/test.yml:4]: yeah just ignore this, I was trying to figure out how to get the `include` in the matrix to work. It sitll doesn't, I'll take another look soon.

as it is configured now, CI runs twice though: one for the push, another for the PR. When sending the PR from a branch inside the repo, all checks show duplicated in the PR status. it's a bit weird,

File: embedded-io-adapters/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+repository = "https://github.com/rust-embedded/embedded-hal"
+readme = "README.md"
+license = "MIT OR Apache-2.0"
+categories = [
+    "embedded",
+    "no-std",
+]
+
+[features]
+std = []
```
Comment (MabezDev) [inline embedded-io-adapters/Cargo.toml:15]:
```suggestion
std = ["embedded-io/std"]
```

The std feature should enable the std feature of embedded-io

Comment (Dirbaio) [reply embedded-io-adapters/Cargo.toml]: done!

File: .github/workflows/test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dirbaio) [reply .github/workflows/test.yml]: Reverted.

**Comment by Dirbaio** [comment]
@eldruin CI issues fixed, please take another look.

i've bumped MSRV to 1.60 to be able to use `dep:`. It'll require changing the required checks  from 1.59 to 1.60 in the repo settings (I don't have permisisons)

**Comment by adamgreig** [comment]
> i've bumped MSRV to 1.60 to be able to use dep:. It'll require changing the required checks from 1.59 to 1.60 in the repo settings (I don't have permisisons)

Have you checked? You should do, you're on the HAL team which has admin access to this repository.

**Comment by Dirbaio** [comment]
True, I now do. I swear I didn't just a few weeks ago :sweat:

**Comment by adamgreig** [comment]
you're not going mad, I fixed it when it was a problem a few weeks ago!

**Review by eldruin — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**